knative/deployer.go

```diff
@@ -3,19 +3,20 @@ package knative
 import (
 	"bytes"
 	"fmt"
-	"io"
-	"os"
+	"sort"
+	"time"
 
-	"knative.dev/client/pkg/kn/root"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/client/pkg/wait"
+	servingv1 "knative.dev/serving/pkg/apis/serving/v1"
+	v1 "knative.dev/serving/pkg/apis/serving/v1"
```

Comment:
You and @dsimansk are taking [different approaches](https://github.com/boson-project/faas/pull/155/files) with this file. I know that you are also dealing with a lot more logic, but I wonder which is right - using the client or using a command?

Reply:
🤷‍♂️

Reply:
We have briefly discussed with @rhuss that it might be beneficial to avoid direct usage of serving/eventing client and rather go with kn abstraction.
For this particular case it'd mean a lot less mangling with service object directly. And in addition we can expect that some level of test coverage is done on kn side already.

Reply:
@dsimansk btw, in this PR we are not using serving/eventing client, but the one provided by knative/client: https://github.com/boson-project/faas/pull/152/files#diff-0cff654ab7f6b7619dd3ebd4a8855f70R11-R37
But I suppose that doesn't change anything from your suggestion, right?

Reply:
I strongly recommend choosing either/or. However, if you use the `kn` API that is offered in front of eventing/serving you still have to import serving/eventing APIs like this, since the signature of the kn client API references the serving objects directly.

For consistencies sake and from an architectural POV though I would definitely recommend to use the client API if it offers all that you need. If not, we should fix this in the client.

Reply:
@rhuss I am starting to be confused a little bit, sorry for that, but I am getting lost in the nomenclature, since the word `client` has multiple meanings in this context 😅 

So your suggestion is to use go client provided byt knative/client? eg:  
```go

client, err := p.NewServingClient(namespace)
...
CreateService(service *servingv1.Service)
```

or to use the kn abstraction as @dsimansk mentioned? eg:
```go
c := service.NewServiceCreateCommand(&params)

args := []string{
	encodedName,
	"--image", f.Image,
	"--env", "VERBOSE=true",
	"--label", "bosonFunction=true",
}
c.SetArgs(args)
err = c.Execute()
```

Reply:
The former. No reference to any cobra commands, but just the services. There is also a better way to create an instance of the service that does not goes over the parameter as in the example above.

Reply:
Mostly to avoid cyclic references, and keep a hierarchical dependency tree.

Reply:
Thanks for the clarification, that's was my original understanding of this.